cupyx/scipy/signal/_splines.py

```diff
@@ -149,13 +233,15 @@ def symiirorder2(input, r, omega, precision=-1.0):
         else:
             precision = 10 ** -cupy.finfo(input.dtype).iexp
 
+    block_sz = 128
     rsq = r * r
     a2 = 2 * r * cupy.cos(omega)
     a3 = -rsq
     cs = cupy.atleast_1d(1 - 2 * r * cupy.cos(omega) + rsq)
+    omega = cupy.asarray(omega, cs.dtype)
+    precision *= precision
 
     # First compute the symmetric forward starting conditions
-    precision *= precision
     pos = cupy.arange(0, input.size + 2, dtype=input.dtype)
     diff = _compute_symiirorder2_fwd_hc(pos, cs, r, omega)
     err = diff * diff
```

Comment:
I tried to move the forward condition computation to its own kernel, but the performance dropped compared w.r.t current diff.